src/components/TodoItem/TodoItem.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+export const TodoItem = React.memo(({ id, title,
+  completed, onCompletedTodos, onRemoveTodo, onTitleChange }) => {
+  const [edit, setEdit] = useState(false);
+  const [newValue, setValue] = useState(title);
+
+  const handleBlur = ({ value }) => {
+    onTitleChange(value, id);
+    setEdit(false);
+  };
+
+  const handleKeyDown = useCallback((e) => {
+    switch (e.key) {
+      case 'Enter': {
+        const { value } = e.target;
+
+        onTitleChange(value, id);
+        setEdit(false);
+        break;
+      }
+
+      case 'Escape': {
+        setEdit(false);
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+  }, [edit]);
+
+  return (
+    <li
+      key={id}
+      className={classNames({ completed, editing: edit })}
```

Comment:
```suggestion
      className={classNames({ completed, editing: edit })}
```
edit is a bad name for boolean, use isEditing